src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -23,16 +22,25 @@
 
 namespace WinRT
 {
+    internal static class KeyValuePairHelper
+    {
+        internal readonly static ConcurrentDictionary<Type, ComInterfaceEntry> KeyValuePairCCW = new();
+
+        internal static void TryAddKeyValuePairCCW(Type keyValuePairType, Guid iid, IntPtr abiToProjectionVftablePtr)
+        {
+            KeyValuePairCCW.TryAdd(keyValuePairType, new ComInterfaceEntry { IID = iid, Vtable = abiToProjectionVftablePtr });
+        }
+    }
+
 #if EMBED
     internal
 #else
     public 
 #endif
     static partial class ComWrappersSupport

```

Comment:
Can't leave a comment for this in that file as it's not in the diff — `Context.CoGetObjectContext` still has a marshalling stub (has a `ref` and an `out` parameter), should we address that as part of this PR? Also related to #1318.

Reply:
Addressed.